fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1707,8 +1727,12 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
                 }
 
                 searchParameter = getSearchParameter(resourceType, parameterName);
-                type = Type.fromValue(searchParameter.getType().getValue());
+                if (searchParameter == null) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(SEARCH_PARAMETER_NOT_FOUND, parameterName, resourceType.getSimpleName()));
+                }
```

Comment:
maybe a decent fit for a small helper method for the 3 places its called?

Reply:
created helper method